thriftbp/client_middlewares.go

```diff
@@ -398,3 +400,28 @@ func PrometheusClientMiddleware(remoteServerSlug string) thrift.ClientMiddleware
 		}
 	}
 }
+
+func getClientError(result thrift.TStruct, err error) error {
+	if err != nil {
+		return err
+	}
+	v := reflect.ValueOf(result)
+	v = reflect.Indirect(v)
+	if v.Kind() != reflect.Struct {
+		return nil
+	}
+	resultField := v.FieldByName("Success").Interface()
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i).Interface()
+		if field == resultField {
+			continue
+		}
+		if tExc, ok := field.(thrift.TException); ok {
+			if tExc == nil || tExc.TExceptionType() != thrift.TExceptionTypeCompiled {
+				continue
+			}
+			return tExc
+		}
```

Comment:
🔕 I think type switches are cleaner, but they take up a bit more vertical space.

```suggestion
		switch t := field.(type) {
			case thrift.TException:
				if tExc == nil || tExc.TExceptionType() != thrift.TExceptionTypeCompiled {
					continue
				}
				return tExc
		}
```